### PR TITLE
Adding Security Rule Examples

### PR DESCRIPTION
**Pull request by matthewdfuller**
*Issue #, if available:* Addresses #18 

*Description of changes:* Adds examples of security checks, covering encryption, backup retention periods, minimum engine versions, TLS protocol versions, and more.

Opening this as a draft first because I intend to add (or welcome someone else to assist with adding) checks for the remaining ~60 supported CloudFormation resource types.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Examples/security_rules.ruleset [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+AWS::CloudTrail::Trail EnableLogFileValidation == true << CloudTrail file validation should be enabled
+AWS::CloudTrail::Trail IncludeGlobalServiceEvents == true << CloudTrail should include global service events
+AWS::CloudTrail::Trail IsLogging == true << CloudTrail logging should be enabled
+AWS::CloudTrail::Trail IsMultiRegionTrail == true << CloudTrail trails should cover all regions
+
+AWS::CodeBuild::Project EncryptionKey == /.*/ << CodeBuild should use a CMK instead of AWS-provided KMS keys
+AWS::CodeBuild::Project VpcConfig.VpcId == /.*/ << CodeBuild projects should be built in a VPC
+
+AWS::CodeBuild::SourceCredential Token == /^\{\{resolve:secretsmanager.*$/ << CodeBuild credentials should use a reference instead of plaintext
```
Comment (PatMyron) [inline Examples/security_rules.ruleset:25]: related discussions of sensitive properties that shouldn't be hardcoded:
https://github.com/aws-cloudformation/cfn-python-lint/pull/1181
https://github.com/stelligent/cfn_nag/issues/253

**Comment by nathanataws** [comment]
This is an excellent contribution.  We're looking at the best way to integrate it with the other examples we're staging up.  Thanks for pulling all of this together!

**Comment by nathanataws** [comment]
@matthewdfuller - Hey, Matthew!  Sorry for the wait.  We just posted our [Example rules policy](https://github.com/aws-cloudformation/cloudformation-guard/blob/master/Examples/README.md).

Would you mind reading through it and confirming things?  I'll do a final read and merge after I get the :thumbsup: from you here.

**Comment by nathanataws** [comment]
I finally had a moment to review these myself.  They appear to meet our Example rules policy with the small exception of not having a "Passing" template.  Given that the rules were written before the policy was published and they meet the rest of the criteria, I'm going to go ahead and merge them.  Thank you again for the help, @matthewdfuller !

File: Examples/security_rules.ruleset [hunk 2]
```diff
@@ -0,0 +1,135 @@
+let backup_retention = 7
```
Comment (PatMyron) [inline Examples/security_rules.ruleset:1]: Retention property discussion here as well:
https://github.com/aws-cloudformation/cfn-python-lint/pull/1634

File: Examples/security_rules.ruleset [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+let allowed_ports = [80, 443]
+
+AWS::AmazonMQ::Broker AutoMinorVersionUpgrade == true << Version upgrades should be enabled to receive security updates
+AWS::AmazonMQ::Broker EncryptionOptions.UseAwsOwnedKey == false << CMKs should be used instead of AWS-provided KMS keys
+AWS::AmazonMQ::Broker EngineVersion == 5.15.10 << Broker engine version should be at least 5.15.10
+AWS::AmazonMQ::Broker Logs.Audit == true << Audit logging should be enabled
+AWS::AmazonMQ::Broker Logs.General == true << General logging should be enabled
+AWS::AmazonMQ::Broker PubliclyAccessible == false << Broker should not be publicly accessible
+
+AWS::AmazonMQ::Configuration EngineVersion == 5.15.10 << Broker engine version should be at least 5.15.10
```
Comment (PatMyron) [inline Examples/security_rules.ruleset:11]: [`MQ 5.15.12`](https://docs.aws.amazon.com/amazon-mq/latest/developer-guide/broker-engine.html) seems even better, but is not allowed by these rules. The last character should probably be a wildcard

File: Examples/security_rules.ruleset [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+AWS::ApiGateway::DomainName SecurityPolicy == TLS_1_2 << API Gateway DomainName SecurityPolicy should use TLS 1.2
+
+AWS::CloudFront::Distribution ViewerCertificate.MinimumProtocolVersion == TLSv1.2_2018 << CloudFront TLS version should be 1.2
```
Comment (PatMyron) [inline Examples/security_rules.ruleset:15]: [`TLSv1.2_2019`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-viewercertificate.html#cfn-cloudfront-distribution-viewercertificate-minimumprotocolversion) seems better, but not allowed by this rule. Last character should probably be a wildcard

**Pull request merged**